core/src/main/scala/akka/persistence/cassandra/journal/TagWriter.scala

```diff
@@ -168,54 +165,55 @@ import scala.util.{ Failure, Success, Try }
     buffer:             Vector[Serialized],
     tagPidSequenceNrs:  Map[String, Long],
     updatedTagProgress: Map[String, Long],
-    awaitingFlush:      Option[ActorRef]   = None
-  ): Receive = {
+    awaitingFlush:      Option[ActorRef]   = None): Receive = {
     case InternalFlush =>
     // Ignore, we will check when the write is done
     case Flush =>
       log.debug("External flush while write in progress. Will flush after write complete")
       become(writeInProgress(buffer, tagPidSequenceNrs, updatedTagProgress, Some(sender())))
     case TagWrite(_, payload) =>
       validate(payload, buffer)
+      if (buffer.size > (4 * settings.maxBatchSize) && System.currentTimeMillis() > (lastLoggedBufferMs + bufferWarningMinDurationMs)) {
+        lastLoggedBufferMs = System.currentTimeMillis()
+        log.warning("Buffer for tagged events is getting too large ({}), is Cassandra responsive? Are writes failing? " +
+          "If events are buffered for longer than the eventual-consistency-delay they won't be picked up by live queries. The oldest event in the buffer is offset: {}", buffer.size, formatOffset(buffer.head.timeUuid))
+      }
       // buffer until current query is finished
-      become(writeInProgress((buffer ++ payload).sortBy(_.timeUuid)(timeUuidOrdering), tagPidSequenceNrs, updatedTagProgress))
+      // Don't sort until the write has finished
+      become(writeInProgress(buffer ++ payload, tagPidSequenceNrs, updatedTagProgress))
     case TagWriteDone(summary, doneNotify) =>
+      val sortedBuffer = buffer.sortBy(_.timeUuid)(timeUuidOrdering)
```

Comment:
When a write is in progress we don't need to check if we should flush so no point sorting until the write is done. 

For when a tagwriter gets behind this made a big difference in load testing.

We could remove the sorting all together and just store min/max timeuuids to know when to flush, I'll raise a ticket for that as this is already a large change.